### PR TITLE
Add `Store` in `VectorCalculator` and functional upgrades

### PR DESCRIPTION
**Pull request by jorgenherje**
Refactor `VectorCalculator` to use app global `Store`-concept and adjust to updates in `VectorSelector`/`SmartNodeSelector`.

Improves usability and simplifies maintenance of component.  Improved layout of component for more consistent view sizes - prevents resizing height when editing expressions or creating new expressions in list - enhanced user experience and usability of component.

---

### Updates:
- Use `Context` and `useReducer` to create a app global `store` for current `state`. 
- Improve layout for consistent view an prevent resizing during editing/usage  
- Component inherit sizes from parent.
- Add dialogs for improved user interaction when selecting new active expression and edited data is not saved
- Updated usage according to new features in `VectorSelector`/`SmartNodeSelector`
- Re-organize folder structure of components for clarity
- Rename components for clarity:
    - ExpressionInputComponent -> `ExpressionEditComponent`
    - VariablesTable->`VectorSelectorTable`

---

### Closes issue:
* https://github.com/equinor/webviz-subsurface-components/issues/608

### CODE REVIEW

**Comment by jorgenherje** [comment]
@hkfb Hi, I had issues with the removed packages in this commit: https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5. When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.

**Comment by hkfb** [comment]
> @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.

Not able to reproduce this. Did you try to remove the `node_modules` directory before running `npm install`?

**Comment by jorgenherje** [comment]
> > @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.
> 
> Not able to reproduce this. Did you try to remove the `node_modules` directory before running `npm install`?

I removed `node_modules` and deleted `package-lock.json` to re-generate everything with `npm install` - this is when I get the issues. If I keep `package-lock.json` and only delete `node_modules` before running `npm install` it seems to work. 

Usually I've been deleting `package-lock.json` and regenerating it with `npm install` rather than performing merge, but now I get issues when doing that.

**Comment by hkfb** [comment]
> @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.

I can reproduce the error when running `npm run typecheck`. I've posted #900 for this. Thanks! :)

**Comment by jorgenherje** [comment]
> > @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.
> 
> I can reproduce the error when running `npm run typecheck`. I've posted #900 for this. Thanks! :)

Great! I guess your `--legacy-peer-deps` in `npm i --ignore-scripts --legacy-peer-deps` is the reason you don't detect it before running `npm run typecheck`? As`--legacy-peer-deps` bypass peerDependency auto-installation as far as I can see? If you use `npm i --ignore-scripts` it would give warning immediately?

**Comment by hkfb** [comment]
> > > @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.
> > 
> > 
> > I can reproduce the error when running `npm run typecheck`. I've posted #900 for this. Thanks! :)
> 
> Great! I guess your `--legacy-peer-deps` in `npm i --ignore-scripts --legacy-peer-deps` is the reason you don't detect it before running `npm run typecheck`? As `--legacy-peer-deps` bypass peerDependency auto-installation as far as I can see? If you use `npm i --ignore-scripts` it would give warning immediately?

As of NPM 7 legacy peer dependencies will simply fail unless you add the `--legacy-peer-deps` argument. In this case it's because of an indirect dependency on legacy React in `@webviz/core-components`

**Comment by hkfb** [comment]
> @hkfb Hi, I had issues with the removed packages in this commit: [3d44653](https://github.com/equinor/webviz-subsurface-components/commit/3d446539ac8bb62966b2ca04cb34489f84ca2eb5). When running `npm install` to re-generate `package-lock.json` I get issues with both Storybook and `store.dispatch()`-call in `DeckGLMap` -component. Seems like `package.json` and `package-lock.json` is off synch, and some dependencies has been updated.

Should be fixed now.

**Review by rubenthoms — CHANGES_REQUESTED** [review]
Good work :+1: Some comments regarding code style and performance.

File: react/src/lib/components/VectorCalculator/components/ExpressionEditComponent/components/VectorSelectorTable.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        );
+
+    const disabled = props.disabled || false;
+
+    const createUpdatedCachedVariableVectorMap = React.useCallback(
+        (newMap: VariableVectorMapType[]): VariableVectorMapType[] => {
+            const newCachedVariableVectorMap = cloneDeep(
+                cachedVariableVectorMap
+            );
+            for (const elm of newMap) {
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply react/src/lib/components/VectorCalculator/components/ExpressionEditComponent/components/VectorSelectorTable.tsx:62]: using `forEach()` as elements of `newMap` contains either new values or edited values for `cachedVariableVectorMap`.

File: react/src/lib/components/VectorCalculator/utils/VectorCalculatorHelperFunctions.ts [hunk 2]
```diff
@@ -233,3 +233,31 @@ export const isVectorNameValid = (
...
+        }
+    }
+    return map;
+};
+
+export const getVariablesFromMap = (
+    variableVectorMap: VariableVectorMapType[]
+): string[] => {
+    const variables = [];
+    for (const elm of variableVectorMap) {
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply react/src/lib/components/VectorCalculator/utils/VectorCalculatorHelperFunctions.ts:259]: Removed function - as it is unused.

File: react/src/lib/components/VectorCalculator/components/ExpressionEditComponent/ExpressionEditComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import { ExpressionInputTextField } from "./components/ExpressionInputTextField";
+
+import {
+    isExpressionEdited,
+    StoreActions,
+    useStore,
+    ExpressionStatus,
+} from "../ExpressionsStore";
+
+import "!style-loader!css-loader!../../VectorCalculator.css";
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply react/src/lib/components/VectorCalculator/components/ExpressionEditComponent/ExpressionEditComponent.tsx:20]: Was updated by Emerson. Will await fix/update in https://github.com/equinor/webviz-subsurface-components/issues/903

File: react/src/lib/components/VectorCalculator/components/ExpressionsStore/ExpressionsStore.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+                return {
+                    ...state,
+                    expressions: newExpressions,
+                    activeExpression: cloneDeep(initialEditableExpression),
+                    editableExpression: initialEditableExpression.expression,
+                    editableName: initialEditableExpression.name,
+                    editableDescription: initialEditableExpression.description,
+                    editableVariableVectorMap: cloneDeep(
+                        initialEditableExpression.variableVectorMap
+                    ),
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply react/src/lib/components/VectorCalculator/components/ExpressionsStore/ExpressionsStore.tsx:191]: Should probably work fine without `cloneDeep`

**Review by rubenthoms — APPROVED** [review]
LGTM, nice improvements :+1:

**Pull request merged**